src/Security/Authentication/OpenIdConnect/src/OpenIdConnectHandler.cs

```diff
@@ -516,6 +552,113 @@ private async Task HandleChallengeAsyncInternal(AuthenticationProperties propert
+    }
+
+    // TODO GetConfigString and ConfigFlagEnabled are only used in PAR, and won't be necessary after
+    // Microsoft.IdentityModel.Protocols.OpenIdConnect is updated to v7.6.1 or later.
+    private bool ConfigFlagEnabled(string name)
+    {
+        if (_configuration?.AdditionalData.TryGetValue(name, out var configValue) ?? false)
+        {
+            if (configValue is bool v)
+            {
+                return v;
+            }
+        }
+        return false;
+    }
+
+    private async Task PushAuthorizationRequest(OpenIdConnectMessage authorizeRequest, AuthenticationProperties properties)
+    {
+        // Build context and run event
+        var parRequest = authorizeRequest.Clone();
+        var context = new PushedAuthorizationContext(Context, Scheme, Options, parRequest, properties);
+        await Events.PushAuthorization(context);
+
+        // If the event handled client authentication, skip the default auth behavior
+        if (context.HandledClientAuthentication)
+        {
+            Logger.PushAuthorizationHandledClientAuthentication();
+        }
+        // Otherwise, add the client secret to the parameters (if available)
+        else
+        {
+            if (!string.IsNullOrEmpty(Options.ClientSecret))
+            {
+                parRequest.Parameters.Add(OpenIdConnectParameterNames.ClientSecret, Options.ClientSecret);
+            }
+        }
+
+        string requestUri;
+
+        // The event can either entirely skip pushing to the par endpoint...
+        if (context.SkippedPush)
+        {
+            Logger.PushAuthorizationSkippedPush();
+            return;
+        }
+        // ... or handle pushing to the par endpoint itself, in which case it will supply the request uri
+        else if (context.HandledPush)
+        {
+            Logger.PushAuthorizationHandledPush();
+            requestUri = context.RequestUri;
+        }
+        else
+        {
+            GetConfigString("pushed_authorization_request_endpoint", out var parEndpoint);
+            if (string.IsNullOrEmpty(parEndpoint))
+            {
+                new InvalidOperationException("Attempt to push authorization with no pushed authorization endpoint configured.");
+            }
+
+            var requestMessage = new HttpRequestMessage(HttpMethod.Post, parEndpoint);
+            requestMessage.Content = new FormUrlEncodedContent(parRequest.Parameters);
+            requestMessage.Version = Backchannel.DefaultRequestVersion;
+            var parResponseMessage = await Backchannel.SendAsync(requestMessage, Context.RequestAborted);
+            requestUri = await GetPushedAuthorizationRequestUri(parResponseMessage);
+        }
+
+        authorizeRequest.Parameters.Clear();
+        authorizeRequest.Parameters.Add("client_id", Options.ClientId);
+        authorizeRequest.Parameters.Add("request_uri", requestUri);
+    }
+
+    // TODO Compare with similar use case in RedeemAuthorizationCodeAsync
```

Comment:
```suggestion
```

Is there anything that `RedeemAuthorizationCodeAsync` that we'd still want to copy. I see that it's less picky about the MediaType, but I like being stricter.

Reply:
I agree, we're good